agent/src/containeroptions/dynamiccontaineroptions.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2016-2017 Pelagicore AB
+ *
+ * Permission to use, copy, modify, and/or distribute this software for
+ * any purpose with or without fee is hereby granted, provided that the
+ * above copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+ * WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR
+ * BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES
+ * OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+ * WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+ * ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
+ * SOFTWARE.
+ *
+ * For further information see LICENSE
+ */
+
+#pragma once
+
+#include "config/softwarecontainerconfig.h"
+
+#include <memory>
+
+namespace softwarecontainer {
+
+class DynamicContainerOptions
+{
+public:
+    DynamicContainerOptions();
```

Comment:
I think the most consistent style here would be to either define the default constructor and destructor and have them completely in the header, or skip both. Since we don't define any other constructor it should be generated for us.